src/evaluate/evaluator/question_answering.py

```diff
@@ -0,0 +1,191 @@
+from ..utils.logging import get_logger
+from .base import Evaluator
+
+
+logger = get_logger(__name__)
+
+
+class QuestionAnsweringEvaluator(Evaluator):
+    """
+    Question answering evaluator.
+    This question answering evaluator can currently be loaded from [`evaluator`] using the default task name
+    `question-answering`.
+    Methods in this class assume a data format compatible with the [`QuestionAnsweringPipeline`].
+    """
+
+    def __init__(self, task="question-answering", default_metric_name=None):
+        super().__init__(task, default_metric_name=default_metric_name)
+
+    def prepare_data(
+        self,
+        data: Union[str, Dataset],
+        question_column: str,
+        context_column: str,
+        id_column: str,
+        label_column: str,
+    ):
+        """Prepare data."""
+        if data is None:
+            raise ValueError(
+                "Please specify a valid `data` object - either a `str` with a name or a `Dataset` object."
+            )
+        data = load_dataset(data) if isinstance(data, str) else data
+        if question_column not in data.column_names:
+            raise ValueError(
+                f"Invalid `question_column` {question_column} specified. The dataset contains the following columns: {data.column_names}."
+            )
+        if context_column not in data.column_names:
+            raise ValueError(
+                f"Invalid `context_column` {context_column} specified. The dataset contains the following columns: {data.column_names}."
+            )
+        if id_column not in data.column_names:
+            raise ValueError(
+                f"Invalid `id_column` {id_column} specified. The dataset contains the following columns: {data.column_names}."
+            )
+        if label_column not in data.column_names:
+            raise ValueError(
+                f"Invalid `label_column` {label_column} specified. The dataset contains the following columns: {data.column_names}."
+            )
+
+        return data
+
+    def compute(
+        self,
+        model_or_pipeline: Union[str, "Pipeline", Callable, "PreTrainedModel", "TFPreTrainedModel"] = None,
+        data: Union[str, Dataset] = None,
+        metric: Union[str, EvaluationModule] = None,
+        tokenizer: Optional[Union[str, "PreTrainedTokenizer"]] = None,
+        strategy: Literal["simple", "bootstrap"] = "simple",
+        confidence_level: float = 0.95,
+        n_resamples: int = 9999,
+        random_state: Optional[int] = None,
+        question_column: str = "question",
+        context_column: str = "context",
+        id_column: str = "id",
+        label_column: str = "answers",
+    ) -> Tuple[Dict[str, float], Any]:
+        """
+        Compute the metric for a given pipeline and dataset combination.
+        Args:
+            model_or_pipeline (`str` or `Pipeline` or `Callable` or `PreTrainedModel` or `TFPreTrainedModel`,
+            defaults to `None`):
```

Comment:
There's a couple of nits in the params documentation here, see https://github.com/huggingface/evaluate/pull/178, I'm fixing them there for the remaining Evaluators.


move line 101 to the previous line